toxencryptsave/toxencryptsave.h

```diff
@@ -165,21 +184,34 @@ bool tox_pass_encrypt(const uint8_t *data, size_t data_len, const uint8_t *passp
 bool tox_pass_decrypt(const uint8_t *data, size_t length, const uint8_t *passphrase, size_t pplength, uint8_t *out,
                       TOX_ERR_DECRYPTION *error);
 
-
-/******************************* BEGIN PART 1 *******************************
+/**
+ * ****************************** BEGIN PART 1 *******************************
  * And now part "1", which does the actual encryption, and is rather less cpu
  * intensive than part one. The first 3 functions are for key handling.
  */
-
-/* This key structure's internals should not be used by any client program, even
+/**
+ * This key structure's internals should not be used by any client program, even
  * if they are straightforward here.
  */
-typedef struct {
-    uint8_t salt[TOX_PASS_SALT_LENGTH];
-    uint8_t key[TOX_PASS_KEY_LENGTH];
-} TOX_PASS_KEY;
+#ifndef TOX_PASS_KEY_DEFINED
+#define TOX_PASS_KEY_DEFINED
+typedef struct Tox_Pass_Key Tox_Pass_Key;
+#endif /* TOX_PASS_KEY_DEFINED */
+
+struct Tox_Pass_Key *tox_pass_key_new(void);
+
+void tox_pass_key_free(struct Tox_Pass_Key *_key);
+
+void tox_pass_key_get_key(const struct Tox_Pass_Key *_key, uint8_t *key);
```

Comment:
No errors are possible? Should this actually read `TOX_PASS_KEY_LENGTH` bytes from given place in memory?
Why there are no docs on how to use it?
If it's something that shouldn't be used, why there's an API for it?

Same goes for other functions.